metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -215,7 +223,11 @@ func NewContainerMetadataEnricher(
 
 				id := join(pod.GetObjectMeta().GetNamespace(), pod.GetObjectMeta().GetName(), container.Name)
 				m[id] = meta
+				if s, ok := statuses[container.Name]; ok {
+					PerfMetrics.ContainerID.SetStringVal(cuid, s.ContainerID)
+				}
```

Comment:
Wondering if we could avoid using the cache for this and use `meta` directly.

Reply:
Yes I will do this! Good catch